prompt/prompt-executor/prompt-executor-clients/prompt-executor-mistralai-client/src/commonMain/kotlin/ai/koog/prompt/executor/clients/mistralai/mapper/MistralAIMessageMapper.kt

```diff
@@ -0,0 +1,63 @@
+package ai.koog.prompt.executor.clients.mistralai.mapper
+
+import ai.koog.prompt.executor.clients.mistralai.model.FunctionCall
+import ai.koog.prompt.executor.clients.mistralai.model.FunctionCallArguments
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIContent
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIMessage
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIMessage.MistralAIAssistantMessage
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIMessage.MistralAISystemMessage
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIMessage.MistralAIToolMessage
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIMessage.MistralAIUserMessage
+import ai.koog.prompt.executor.clients.mistralai.model.MistralAIToolCall
+import ai.koog.prompt.message.Message
+
+internal object MistralAIMessageMapper {
+
+    fun mapToMistralAIMessage(message: Message): MistralAIMessage {
+        return when (message) {
+            is Message.System -> message.toMistralAIMessage()
+            is Message.User -> message.toMistralAIMessage()
+            is Message.Assistant -> message.toMistralAIMessage()
+            is Message.Tool.Result -> message.toMistralAIMessage()
+            is Message.Tool.Call -> message.toMistralAIMessage()
+        }
+    }
+}
+
+private fun createTextContent(text: String): List<MistralAIContent.ContentChunk.TextChunk> =
+    if (text.isNotEmpty()) listOf(MistralAIContent.ContentChunk.TextChunk(text)) else emptyList()
+
+private fun Message.User.toMistralAIMessage(): MistralAIUserMessage {
+    return MistralAIUserMessage(content = createTextContent(content))
```

Comment:
Could you please consider supporting media types as well? 

Mistral supports images and media (ex: https://docs.mistral.ai/capabilities/vision/ )